core/src/main/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationService.scala

```diff
@@ -109,4 +110,21 @@ class GoogleProjectRegistrationService(protected val ctx: RawlsRequestContext,
         case None => Future.successful(None)
       }
     } yield finalResult
+
+  def unregisterGoogleProject(googleProjectId: GoogleProjectId): Future[Unit] =
+    // 1. Check delete action on google-project resource.
+    for {
+      _ <- samDAO
+        .userHasAction(SamResourceTypeNames.googleProject, googleProjectId.value, SamGoogleProjectActions.delete, ctx)
+        .map(canDelete =>
+          if (!canDelete)
+            throw new RawlsExceptionWithErrorReport(errorReport =
+              ErrorReport(StatusCodes.Forbidden, s"Google project not found or you do not have permission to delete.")
+            )
+        )
+      // 2. Disable billing on the Google project.
+      _ <- googleServicesDAO.disableBillingOnGoogleProject(googleProjectId, ctx.toTracingContext)
```

Comment:
is this fast enough to do in a synchronous API? I think it will be based on what I've seen using `gcloud`, but figured I'd ask …

Reply:
I think so, it took maybe a second when I tried it out.